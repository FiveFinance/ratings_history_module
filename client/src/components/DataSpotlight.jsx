import React from 'react';
import styled from 'styled-components';
import { Tag } from 'styled-icons/fa-solid/';

class DataSpotlight extends React.Component {
  constructor(props) {
    super(props);
    this.buy = 3;
    this.hold = 2;
    this.sell = 8;
    this.total = this.buy + this.hold + this.sell;
    this.percentage = Math.floor(100 * this.buy / this.total);
  }

  // eslint-disable-next-line class-methods-use-this
  imgError(image) {
    image.onerror = '';
    image.src = '../../dist/lib/tag.png';
  }

  render() {
    return (
      <Circle>
        <TagPercent>
          <FlippedTag />
          &nbsp;
          {this.percentage}
          %
        </TagPercent>
        <OfNRatings>
          of {this.total} ratings
        </OfNRatings>
      </Circle>
    );
  }
}

const Circle = styled.div`
  width: 134px;
  height: 134px;
  border-radius: 50%;
  background-color: rgba(33, 206, 153, 0.30);
  display: flex;
  justify-content: center;
  flex-direction: column;
  align-items: center;
  
`;

const TagPercent = styled.div`
  font-family: "DIN Pro", -apple-system, system-ui, sans-serif;
  font-weight: normal;
  -webkit-text-stroke: 1px;
  font-size: 26px;
  line-height: 30px;
  text-align: center;
  letter-spacing: -0.14px;
  color: #21ce99;
  display: flex;
`;

const FlippedTag = styled(Tag)`
  width: 16px;
  height: 16px;
  display: flex;
  align-self: center;
  color: #21ce99;
  
  -moz-transform: scaleX(-1);
  -o-transform: scaleX(-1);
  -webkit-transform: scaleX(-1);
  transform: scaleX(-1);
  filter: FlipH;
  -ms-filter: "FlipH";
  display: flex;
`;

const OfNRatings = styled.div`
  font-weight: normal;
  font-size: 13px;
  -webkit-text-stroke: 0.8px;
  letter-spacing: 0.40px;
  display: flex;
  color: #21ce99;
`;

export default DataSpotlight;


{/* 
  import tagSVG from '../../dist/lib/tag.svg';
  <img src={tagSVG} alt="price tag" width="14px" height="14px" onError={this.imgError(this)} />
{this.percentage}
% 
*/}