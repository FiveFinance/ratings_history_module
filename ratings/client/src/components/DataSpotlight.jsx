import React from 'react';
import styled, { css } from 'styled-components';
import PropTypes from 'prop-types';
// import { Tag } from 'styled-icons/fa-solid/';
import svgTagBEAR from './tagBEAR.svg';
import svgTagBULL from './tagBULL.svg';

const BULL = '#21ce99';
const BEAR = '#f45531';
const BULLCIRCLE = 'rgba(33, 206, 153, 0.15)';
const BEARCIRCLE = 'rgba(244, 85, 49, 0.15)';

const DataSpotlight = ({ stock, market, updateMarket }) => (
  <Circle market={market} onClick={updateMarket}>
    <TagPercent market={market}>
      {market === 'Bear' && <img src={svgTagBEAR} alt="price tag" width="18px" height="16px" />}
      {market === 'Bull' && <img src={svgTagBULL} alt="price tag" width="18px" height="16px" />}
      {`  ${Math.floor(100 * stock.recBuy / (stock.recBuy + stock.recHold + stock.recSell))}%`}
    </TagPercent>
    <OfNRatings market={market}>
      {`of ${stock.recBuy + stock.recHold + stock.recSell} ratings`}
    </OfNRatings>
  </Circle>
);

const Circle = styled.button`
  width: 134px;
  height: 134px;
  border-radius: 50%;
  background-color: ${({ market }) => (market === 'Bear' ? BEARCIRCLE : BULLCIRCLE)};
  display: flex;
  justify-content: center;
  flex-direction: column;
  align-items: center;

  -webkit-appearance: inherit;
  -webkit-appearance: initial;
  padding: 0;
  border: none;
  
  &:focus {
    outline:0;
  }
`;

const TagPercent = styled.div`
  font-family: "DIN Pro", -apple-system, system-ui, sans-serif;
  font-weight: normal;
  -webkit-text-stroke: 1px;
  font-size: 26px;
  line-height: 36px;
  text-align: center;
  letter-spacing: -0.14px;
  color: ${({ market }) => (market === 'Bear' ? BEAR : BULL)};
  display: flex-inline;
  align-items: center;
`;

// const FlippedTag = styled(Tag)`
//   width: 16px;
//   height: 16px;
//   display: flex;
//   align-self: center;
//   color: #21ce99;
//   -moz-transform: scaleX(-1);
//   -o-transform: scaleX(-1);
//   -webkit-transform: scaleX(-1);
//   transform: scaleX(-1);
//   filter: FlipH;
//   -ms-filter: "FlipH";
//   display: flex;
// `;

const OfNRatings = styled.div`
  font-weight: normal;
  font-size: 13px;
  -webkit-text-stroke: 0.8px;
  letter-spacing: 0.0350em;
  display: flex;
  color: ${({ market }) => (market === 'Bear' ? BEAR : BULL)};
`;

DataSpotlight.propTypes = {
  stock: PropTypes.shape({
    buy: PropTypes.number,
    hold: PropTypes.number,
    sell: PropTypes.number,
  }).isRequired,
  market: PropTypes.string.isRequired,
  updateMarket: PropTypes.func.isRequired,
};

export default DataSpotlight;
