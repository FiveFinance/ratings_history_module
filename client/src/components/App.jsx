import React from 'react';
import DataSpotlight from './DataSpotlight';
import DataBarChart from './DataBarChart';
import AnalystReviews from './AnalystReviews';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      stocks: 'placeholder',
      purchases: 'otherplaceholder',
    };
    this.updateStocks = this.updateStocks.bind(this);
    this.updatePurchases = this.updatePurchases.bind(this);
  }

  componentDidMount() {
    this.updateStocks('hi');
    this.updatePurchases('bye');
  }

  updateStocks(data) {
    this.setState = ({ stocks: data });
  }

  updatePurchases(data) {
    this.setState = ({ purchases: data });
  }

  render() {
    const { state } = this.state;
    //const { stocks } = state.stocks;
    //const { purchases } = state.purchases;

    return (
      <div>
        <h1>Robinhood Expert Ratings Module</h1>

        <DataSpotlight />
        <DataBarChart />
        <AnalystReviews />
      </div>
    );
  }
}

export default App;
