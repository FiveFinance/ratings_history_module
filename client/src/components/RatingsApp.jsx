import React from 'react';
import styled from 'styled-components';
import DataSpotlight from './DataSpotlight';
import DataBarChart from './DataBarChart';
import ReviewList from './ReviewList';

class RatingsApp extends React.Component {
  constructor() {
    super();
    this.state = {
      stock: {
        symbol: 'TSM',
        recBuy: 3,
        recHold: 8,
        recSell: 13,
        reviewBuy: 'Plastic granular scale convergence Taiwan Semiconductor Manufacturing Company Limited local Ergonomic. \n The impactful transform architectures encompassing. \n Overall, granular scale convergence Taiwan Semiconductor Manufacturing Company Limited optimizing',
        reviewSell: 'Plastic B2C target e-tailers local Taiwan Semiconductor Manufacturing Company Limited Ergonomic. \n For B2C implement applications encompassing. \n Hence, B2C target e-tailers Taiwan Semiconductor Manufacturing Company Limited optimizing',
      },
      priceDifferenceFromStartToToday: 'Positive',
    };
  }

  render() {
    return (
      <div>
        <ModuleHeader>Analyst Ratings</ModuleHeader>
        <Top>
          <DataSpotlight stock={this.state.stock} priceDifferenceFromStartToToday={this.state.priceDifferenceFromStartToToday} />
          <DataBarChart stock={this.state.stock} priceDifferenceFromStartToToday={this.state.priceDifferenceFromStartToToday} />
        </Top>
        <ReviewList stock={this.state.stock} />
      </div>
    );
  }
}

const Top = styled.div`
  display: flex;
  flex: 1 100%;
  flex-shrink: 1;
`;

const ModuleHeader = styled.div`
  font-family: "DIN Pro", -apple-system, system-ui, sans-serif;
  font-weight: bold;
  font-size: 26px;
  line-height: 30px;
  text-align: start;
  letter-spacing: -0.14px;
  border-bottom: 1px solid;
  border-color: #f4f4f5;
  padding-bottom: 16px;
`;

export default RatingsApp;
