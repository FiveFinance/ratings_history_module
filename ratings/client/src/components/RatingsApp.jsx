import React from 'react';
import styled from 'styled-components';
import axios from 'axios';
import DataSpotlight from './DataSpotlight';
import DataBarChart from './DataBarChart';
import ReviewList from './ReviewList';

class RatingsApp extends React.Component {
  constructor() {
    super();
    this.state = {
      stock: {
        symbol: 'PHDR',
        company: 'Placeholder Company',
        recBuy: 18,
        recHold: 18,
        recSell: 18,
        reviewBuy: 'Rubber killer recontextualize web services Placeholder impactful Gorgeous. \n The viral envisioneer initiatives Et perspiciatis error quas.. \n Overall, killer recontextualize web services Microsoft Corporation reciprocal Doloribus numquam expedita sed fugit tempore in adipisci.',
        reviewSell: 'Rubber open-source envisioneer channels impactful Placeholder Gorgeous. \n For sexy strategize bandwidth Et perspiciatis error quas.. \n Hence, open-source envisioneer channels Microsoft Corporation reciprocal Doloribus numquam expedita sed fugit tempore in adipisci.',
      },
      market: 'Bull',
    };
    // this.market = this.state.pricedifferenceFromStartToToday === 'Negative' ? 'Bear' : 'Bull';
    this.getStockData = this.getStockData.bind(this);
    this.updateMarket = this.updateMarket.bind(this);
  }

  componentDidMount() {
    this.getStockData();
  }

  getStockData() {
    const stockID = window.location.pathname.split('/')[2];
    if (!stockID) {
      axios.get('/api/ratings/AAPL')
        .then(res => res.data)
        .then((result) => {
          this.setState({
            stock: result,
          });
        });
    } else {
      axios.get(`/api/ratings/${stockID}`)
        .then(res => res.data)
        .then((result) => {
          this.setState({
            stock: result,
          });
        });
    }
  }

  updateMarket(e) {
    const { market } = this.state;
    e.preventDefault();
    this.setState({
      market: market === 'Bull' ? 'Bear' : 'Bull',
    });
  }

  render() {
    const { stock, market } = this.state;
    return (
      <div>
        <ModuleHeader>Analyst Ratings</ModuleHeader>
        <Row>
          <Left>
            <DataSpotlight stock={stock} market={market} updateMarket={this.updateMarket} />
          </Left>
          <Right>
            <DataBarChart stock={stock} market={market} />
            <ReviewList stock={stock} market={market} />
          </Right>
        </Row>
      </div>
    );
  }
}

const Row = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  margin-top: 16px;
`;

const Left = styled.div`
  width: 134px;
  padding: 13px 48px 24px 24px;
`;

const Right = styled.div`
  display: flex;
  flex-direction: column;
  width: 480px;
`;

const ModuleHeader = styled.div`
  font-family: "DIN Pro", -apple-system, system-ui, sans-serif;
  font-weight: normal;
  font-size: 26px;
  line-height: 30px;
  text-align: start;
  letter-spacing: -0.14px;
  border-bottom: 1px solid;
  border-color: #f4f4f5;
  padding-bottom: 16px;
  max-width: 697px;
`;

export default RatingsApp;
