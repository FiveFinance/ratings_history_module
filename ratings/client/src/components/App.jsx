import React from 'react';
import styled from 'styled-components';
import RatingsApp from './RatingsApp';

const HistoryApp = () => (
  <Wrapper>
    <RatingsApp />
  </Wrapper>
);

const Wrapper = styled.div`
  width: 700px + .05vw;
  padding-left: 40px;
  -webkit-transition: background 300ms;
  transition: background 300ms;

  @font-face {
    font-family: DIN Pro;
    font-weight: normal;
    src: url("fonts/DINProRegular.otf") format("opentype");
  }

  body, p, div {
    font-family: "DIN Pro", -apple-system, system-ui, sans-serif;
    font-weight: normal;
  }
`;

export default HistoryApp;
