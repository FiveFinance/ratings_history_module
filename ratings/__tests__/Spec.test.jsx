import React from 'react';
import { shallow, mount, render } from 'enzyme';
import sinon from 'sinon';
// import { expect } from 'chai';
import App from '../client/src/components/App';
import RatingsApp from '../client/src/components/RatingsApp';
import DataSpotlight from '../client/src/components/DataSpotlight';

global.React = React;
global.shallow = shallow;
global.mount = mount;
global.sinon = sinon;

function setup() {
  const myApp = shallow(<App />);
  const ratingsApp = shallow(<RatingsApp />);
  return { myApp, ratingsApp };
}

it('should render ratings correctly', () => {
  const wrapper = render(
    <RatingsApp />
  );
  expect(RatingsApp).toMatchSnapshot();
});

describe("Ratings App Module", () => {
  test("renders", () => {
    const wrapper = shallow(<RatingsApp />);
    expect(wrapper.exists()).toBe(true);
  })
});

const dummyData = [{
  stock: {
    symbol: 'TST',
    company: 'Fake Company Inc',
    recBuy: 11,
    recHold: 15,
    recSell: 18,
    reviewBuy: 'Rubber killer recontextualize web services Microsoft Corporation impactful Gorgeous. \n The viral envisioneer initiatives Et perspiciatis error quas.. \n Overall, killer recontextualize web services Microsoft Corporation reciprocal Doloribus numquam expedita sed fugit tempore in adipisci.',
    reviewSell: 'Rubber open-source envisioneer channels impactful Microsoft Corporation Gorgeous. \n For sexy strategize bandwidth Et perspiciatis error quas.. \n Hence, open-source envisioneer channels Microsoft Corporation reciprocal Doloribus numquam expedita sed fugit tempore in adipisci.',
  },
  market: 'Bull',
}]

const mock = dummyData[0];
