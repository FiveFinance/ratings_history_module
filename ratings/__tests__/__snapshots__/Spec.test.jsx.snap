// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ratings correctly 1`] = `[Function]`;
